### ISSUE TITLE
Play scene stuck on the last played scene.

### ISSUE THREAD

**Issue by raditwbw** [NibbleByte/UnityAssetManagementTools#1]
This might be not an issue, because I already found the solution. But you might consider adding an option.

After pressing the `PlaySceneDirectly` button, whenever I press the normal play button at the top, it will always play the last scene that played using the `PlaySceneDirectly` button.

It is because `EditorSceneManager.playModeStartScene` is not reset to default.
My solution, for now, is to add this class

```
[InitializeOnLoad]
public class PlaymodeStateTool
{
    static PlaymodeStateTool()
    {
        EditorApplication.playModeStateChanged += PlaymodeStateChanged;
    }

    private static void PlaymodeStateChanged(PlayModeStateChange state)
    {
        if (state == PlayModeStateChange.ExitingPlayMode)
        {
            EditorSceneManager.playModeStartScene = default;
        }
    }
}
```

But maybe you want to add an option to the setting to reset the `EditorSceneManager.playModeStartScene` to default.

**Comment by NibbleByte** [comment]
That is a great catch! I never would have found this, thanks! Will fix it soon.

**Comment by NibbleByte** [comment]
I've made a simpler fix, you can check my commit (3cd4145922fc3540278bc5a4b09ab15f7fc340bc). When you run `PlaySceneDirectly()` it sets a flag in the window instance indicating it needs to clear the `EditorSceneManager.playModeStartScene`, then clears it in `OnEnable()`. It is guaranteed to be called, since you started play mode from that very window and unless some other magic closes it until the scene is loaded, it should work.
 
Then I realized that you can now disable assembly reload so `OnEnabled()` won't get called then. 😄 So I had to ensure the code gets called in that case as well - 5ee076bb5fa11417ce13b1f4c245a01c5c7a93d9.

This way it will clear the field only if the window instance used it. It will also restore the original value of the property, in case some other tool was using it - d2e8b5ba0f323f00b03d7e94ab7aebb13662281c.

Will close the issue and soon publish the changes to the Asset Store. UPM is good to go. Let me know if this fix doesn't work for you.
